shop/src/main/java/ro/msg/learning/shop/config/security/WebSecurityFormConfig.java

```diff
@@ -0,0 +1,50 @@
+package ro.msg.learning.shop.config.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+@Configuration
+@EnableWebSecurity
+@Profile({"with-form"})
+public class WebSecurityFormConfig extends WebSecurityConfigurerAdapter {
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .antMatchers("/login").permitAll()
+                .antMatchers("/product/get_all_products").permitAll()
+                .antMatchers("/product/**").authenticated()
+                .antMatchers("/order/**").authenticated()
+                .and()
+                .formLogin()
+                .loginPage("/login")
+                .defaultSuccessUrl("/product/get_all_products", true)
+                .failureUrl("/login-error");
+
+        http.csrf().csrfTokenRepository(CookieCsrfTokenRepository.withHttpOnlyFalse());
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
```

Comment:
A tip here: if you override the configure method and get rid of the **@Autowired** configureGlobal one, your circular dependency will be fixed. (https://stackoverflow.com/questions/35218354/difference-between-registerglobal-configure-configureglobal-configureglo)